tools/credentials.py

```diff
@@ -0,0 +1,43 @@
+import os, argparse, json
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.oauth2.credentials import Credentials
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Obtains credentials for youtube separate from upload.py")
+
+    parser.add_argument("-c", "--client", default="client_secrets.json", help="Path to client secrets file. Required if token is not valid (default: %(default)s)")
+    parser.add_argument("-t", "--token", default="credentials.json", help="Path to authorization token file. If does not exist or invalid, this script will take you through the Google OAuth process (default: %(default)s)")
+    parser.add_argument("--no-save-token", dest="save_token", action="store_false", help="Disable saving the authorization token to the path specified in (--token), after authorization.")
+    return parser.parse_args()
+
+def main():
+    args = parse_args()
+
+    if os.path.isfile(args.token):
```

Comment:
If you put everything from here down to `json.dump` to a separate function, then `upload.py` can reuse it wholesale. Currently this part is exactly the same.